p2p/conn/connection.go

```diff
@@ -802,15 +804,22 @@ func (ch *Channel) SetLogger(l log.Logger) {
 	ch.Logger = l
 }
 
-// Queues message to send to this channel.
+// sendBytes queues a message to send to this channel.
 // Goroutine-safe
 // Times out (and returns false) after defaultSendTimeout.
 func (ch *Channel) sendBytes(bytes []byte) bool {
+	// Reset timer instead of creating a new one each time
+	ch.timer.Reset(defaultSendTimeout)
+
 	select {
 	case ch.sendQueue <- bytes:
 		atomic.AddInt32(&ch.sendQueueSize, 1)
 		return true
-	case <-time.After(defaultSendTimeout):
+	case <-ch.timer.C:
+		// Stop the timer to release its resources
+		if !ch.timer.Stop() {
```

Comment:
If we stop the timer, it will never fire again. This means that upon the next send, we will potentially block forever because there will be no timeout.

Reply:
Reset(..) will reset it